pillar/base/users.sls

```diff
@@ -251,8 +248,6 @@ users:
       mail:
         allowed: True
         sudo: True
-        groups:
-          - mailman
```

Comment:
We want to leave the `mailman` group for users on mail.python.org which also runs mailman. The `mailman` role was back when we had GNU mailman project infra hosted inside the PSF infra long ago.